### PR TITLE
feat: navigationstepper component

### PR DESCRIPTION
**Pull request by DorianMaliszewski**
## Type

- Feature

#### What is expected?

New Navigation Stepper (or Progress) component

#### The following changes where made:

1. The component

2. Stories

3. Tests

### CODE REVIEW

**Review by chambo-e — APPROVED** [review]

File: src/components/NavigationStepper/__tests__/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    shouldMatchEmotionSnapshot(
+      <NavigationStepper condensed step={2}>
+        <NavigationStepper.Step isLoading>First</NavigationStepper.Step>
+        <NavigationStepper.Step>Second</NavigationStepper.Step>
+        <NavigationStepper.Step>Third</NavigationStepper.Step>
+      </NavigationStepper>,
+    ))
+
+  test('should render correctly with default condensed', async () => {
+    Object.defineProperty(window, 'matchMedia', {
```
Comment (chambo-e) [inline src/components/NavigationStepper/__tests__/index.test.tsx:57]: Already done in beforeAll ?

File: src/utils/responsive/useMedia.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { useCallback, useEffect, useState } from 'react'
```
Comment (chambo-e) [inline src/utils/responsive/useMedia.ts:1]: https://github.com/streamich/use-media

https://github.com/jepser/use-match-media

https://github.com/streamich/react-use/blob/master/docs/useMedia.md


Plenty of options ?

File: src/components/NavigationStepper/index.tsx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+  step = 1,
+}) => {
+  const theme = useTheme()
+  const isMobile = useMedia(
+    [`(max-width: ${theme.screens.small}px)`],
+    [true],
+    false,
+  )
+
+  console.log('IsMobile', isMobile)
```
Comment (chambo-e) [inline src/components/NavigationStepper/index.tsx:161]: 💩

File: src/components/NavigationStepper/index.tsx [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+            <StyledStepContent aria-hidden={!isCurrent && computedCondensed}>
+              {child}
+            </StyledStepContent>
+          </StyledStep>
+        )
+      }
+
+      return null
+    },
+  ) as ReactNode[]
```
Comment (chambo-e) [inline src/components/NavigationStepper/index.tsx:206]: If you filter out null values would you still need to cast as ReactNode[] ?

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: src/components/NavigationStepper/__tests__/index.test.tsx (hunk 1 above)
Comment (DorianMaliszewski) [reply src/components/NavigationStepper/__tests__/index.test.tsx:57]: But I switch from matches from false to true to emulate the default condensed at true for small screens

File: src/utils/responsive/useMedia.ts (hunk 2 above)
Comment (DorianMaliszewski) [reply src/utils/responsive/useMedia.ts:1]: I took the one from react-hooks, do you prefer to add a dependency ?

File: src/components/NavigationStepper/index.tsx (hunk 4 above)
Comment (DorianMaliszewski) [reply src/components/NavigationStepper/index.tsx:206]: No it is `steps.length` the problem cause React.Children.map can return undefined

File: src/utils/responsive/useMedia.ts (hunk 2 above)
Comment (chambo-e) [reply src/utils/responsive/useMedia.ts:1]: yes, less maintenance for us :)

File: src/components/NavigationStepper/index.tsx (hunk 4 above)
Comment (chambo-e) [reply src/components/NavigationStepper/index.tsx:201]: Alright :)

**Review by chambo-e — APPROVED** [review]

**Review by philibea — APPROVED** [review]

**Pull request merged**